avro-schema-processor/src/main/java/io/micronaut/avro/visitor/AvroSchemaVisitor.java

```diff
@@ -0,0 +1,433 @@
+                .refType(true)
+                .build();
+        }
+        // Handle simple types directly
+        if (classElement.isAssignable(Number.class) || classElement.isPrimitive()) {
+            AvroSchema primitiveSchema = new AvroSchema();
+            setSchemaType(element, visitorContext, context, primitiveSchema);
+            return primitiveSchema;
+        }
+
+        return createTopLevelSchema(element, visitorContext, context);
+    }
+
+    private static void setSchemaType(TypedElement element, VisitorContext visitorContext, AvroSchemaContext context, AvroSchema avroSchema) {
+        ClassElement type = element.getGenericType();
+
+        if (type.isAssignable(Map.class)) {
+            avroSchema.setType(Type.MAP);
+            ClassElement valueType = type.getTypeArguments().get("V");
+            AvroSchema valueSchema = createSchema(valueType, visitorContext, context);
+            // Simplify primitive values in maps
+            if (isPrimitiveType(valueType) || isPrimitiveAvroType(valueSchema.getType())) {
+                avroSchema.setValues(valueSchema.getType());
+            } else {
+                avroSchema.setValues(valueSchema);
+            }
+        } else if (type.isAssignable(Collection.class)) {
+            avroSchema.setType(Type.ARRAY);
+            avroSchema.setJavaClass(type.getCanonicalName());
+            ClassElement componentType = type.getFirstTypeArgument().orElse(null);
+            if (componentType != null) {
+                AvroSchema itemSchema = createSchema(componentType, visitorContext, context);
+                // If the component type is primitive, use the simple type string
+                if (isPrimitiveType(componentType) || isPrimitiveAvroType(itemSchema.getType())) {
+                    avroSchema.setItems(itemSchema.getType());
+                } else {
+                    avroSchema.setItems(itemSchema);
+                }
+            }
+        } else if (!type.isPrimitive() && type.getRawClassElement() instanceof EnumElement enumElement) {
+            avroSchema.setType(Type.ENUM);
+            avroSchema.setName(element.getSimpleName());
+            for (String value : enumElement.values()) {
+                avroSchema.addSymbol(value);
+            }
+            context.currentOriginatingElements().add(enumElement);
+        } else if (isPrimitiveType(type)) {
+            // todo
+            avroSchema.setType(getPrimitiveType(avroSchema, type));
+
+
+        } else if (type.isAssignable(Number.class)) {
+            setNumericType(type, avroSchema, context);
+        } else if (type.isAssignable(CharSequence.class)) {
+            if (type.getName().equals("java.lang.Character") || type.getName().equals("char")) {
+                avroSchema.setUnsupported(true);
+                avroSchema.setType(Type.INT);
+                avroSchema.setJavaClass("java.lang.Character");
+            } else {
+                avroSchema.setType(Type.STRING);
```

Comment:
Perhaps if there is `@jakarta.validation.Size(min=value, max=value)` it could set the type to `Type.FIXED`.

Reply:
future work